level_handler.go

```diff
@@ -226,44 +236,48 @@ func (s *levelHandler) getTableForKey(key []byte) ([]*table.Table, func() error)
 			out = append(out, s.tables[i])
 			s.tables[i].IncrRef()
 		}
-		return out, func() error {
-			for _, t := range out {
-				if err := t.DecrRef(); err != nil {
-					return err
-				}
-			}
-			return nil
-		}
+		s.RUnlock()
+		return out
 	}
 	// For level >= 1, we can do a binary search as key range does not overlap.
 	idx := sort.Search(len(s.tables), func(i int) bool {
 		return y.CompareKeys(s.tables[i].Biggest(), key) >= 0
 	})
 	if idx >= len(s.tables) {
 		// Given key is strictly > than every element we have.
-		return nil, func() error { return nil }
+		s.RUnlock()
+		return nil
 	}
 	tbl := s.tables[idx]
 	tbl.IncrRef()
-	return []*table.Table{tbl}, tbl.DecrRef
+	s.RUnlock()
+	return []*table.Table{tbl}
 }
 
 // get returns value for a given key or the key after that. If not found, return nil.
 func (s *levelHandler) get(key []byte) (y.ValueStruct, error) {
-	tables, decr := s.getTableForKey(key)
+	tables := s.getTableForKey(key)
+	defer func() {
+		for _, t := range tables {
+			if err := t.DecrRef(); err != nil {
+				panic(err)
+			}
+		}
+	}()
+
 	keyNoTs := y.ParseKey(key)
 
 	var maxVs y.ValueStruct
 	for _, th := range tables {
 		if th.DoesNotHave(keyNoTs) {
-			y.NumLSMBloomHits.Add(s.strLevel, 1)
+			//y.NumLSMBloomHits.Add(s.strLevel, 1)
```

Comment:
We don't need the stats?

Reply:
We will redesign the stats show.

Reply:
Can we just delete the line instead of comment out?